pkg/didcomm/protocol/introduce/service.go

```diff
@@ -348,7 +354,13 @@ func (s *Service) newDIDCommActionMsg(md *metaData) service.DIDCommAction {
 
 			s.processCallback(md)
 		},
-		Stop: func(err error) { actionStop(customError{error: err}) },
+		Stop: func(err error) {
+			if err == nil {
+				err = errProtocolStopped
+			}
+
+			actionStop(customError{error: err})
```

Comment:
if err != nil, the Stop action doesn't include errProtocStopped, ie the stop error is hidden, is this the intended behaviour?

Reply:
when we stop the protocol we need to provide an error (reason), if the user doesn’t provide it we will use default message e.g `errProtocolStopped`. It is not about behavior, it is more as info.